### PR TITLE
Implement Progress Bar for Install

### LINKED ISSUE #518: Install-PSResource: implement progress bar
For Install-PSResource we still need to implement the progress bar

### PR DESCRIPTION
**Pull request by anamnavi**
# PR Summary

Add Progress Bar for Install-Helper. This means that cmdlets which call into Install-Helper (i.e. Update/Save) will now also have a progress bar displayed while the install portion of the operation is occurring.

Not sure how to capture the progress bar in tests, but I tested interactively and checked the progress bar for the following scenarios,
where PSGetTestModule has a dependency package, PSGetTestDependency1.

Scenarios:
1) `Install-PSResource -Name "PSGetTestModule" -Repository PoshTestGallery`
2) `Install-PSResource -Name "PSGetTestModule","TestModule" -Repository PoshTestGallery`
3) scenario where the parent package is installed (and won't reinstall) but dependency package needs to be installed:
`Install-PSResource -Name "PSGetTestModule" -Repository PoshTestGallery`
`Uninstall-PSResource -Name "PSGetTestDependency1 -Force`
`Install-PSResource -Name "PSGetTestModule" -Repository PoshTestGallery`

Note: We already have tests for installing PSGetTestModule and checking for its dependency package.

## PR Context

Part of missing cmdlet implementation, resolves #518 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by alerickson — APPROVED** [review]
Looks good to me!

File: src/code/InstallHelper.cs [hunk 1]
```diff
@@ -218,6 +218,7 @@ public InstallHelper(bool savePkg, PSCmdlet cmdletPassedIn)
 
                 List<string> pkgsInstalled = InstallPackage(
                     pkgsFromRepoToInstall,
+                    pckgNamesToInstall,
```
Comment (PaulHigin) [inline src/code/InstallHelper.cs]: Nitpick.  We should use consistent naming ... 'pkgNamesToInstall'.

File: src/code/InstallHelper.cs [hunk 2]
```diff
@@ -279,15 +280,25 @@ private IEnumerable<PSResourceInfo> FilterByInstalledPkgs(IEnumerable<PSResource
...
-            foreach (PSResourceInfo pkgInfo in pkgsToInstall)
+            int totalPkgs = pkgsToInstall.Count();
+
+            // by default this is 1, because if a parent package was already installed and only the dependent package
+            // needs to be installed we don't want a default value of 0 which throws a division error.
+            // if parent package isn't already installed we'll set this value properly in the below if condition anyways
+            int currentPkgNumOfDependentPkgs = 1;
+            // counter for dependency packages
+            int j = 1;
+            for (int i = 0; i < pkgsToInstall.Count(); i++)
```
Comment (PaulHigin) [inline src/code/InstallHelper.cs]: It is inefficient to pass in an enumerator and then use it like an array type.  Here, I think we can use foreach, and create a new 'count' variable outside the loop to keep track of the install package number.

File: src/code/InstallHelper.cs [hunk 3]
```diff
@@ -279,15 +280,25 @@ private IEnumerable<PSResourceInfo> FilterByInstalledPkgs(IEnumerable<PSResource
...
             List<string> pkgsSuccessfullyInstalled = new List<string>();
-            foreach (PSResourceInfo pkgInfo in pkgsToInstall)
+            int totalPkgs = pkgsToInstall.Count();
+
+            // by default this is 1, because if a parent package was already installed and only the dependent package
+            // needs to be installed we don't want a default value of 0 which throws a division error.
+            // if parent package isn't already installed we'll set this value properly in the below if condition anyways
+            int currentPkgNumOfDependentPkgs = 1;
+            // counter for dependency packages
+            int j = 1;
```
Comment (PaulHigin) [inline src/code/InstallHelper.cs]: Please use descriptive variable names:
```csharp
int totalPkgCount = 0;
int dependentPkgCount = 1

foreach (var pkg in pkgsToInstall)
{
    totalPkgCount++;
    ...
}
```

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/code/InstallHelper.cs [hunk 4]
```diff
@@ -279,15 +280,25 @@ private IEnumerable<PSResourceInfo> FilterByInstalledPkgs(IEnumerable<PSResource
         }
 
         private List<string> InstallPackage(
-            IEnumerable<PSResourceInfo> pkgsToInstall,
+            IEnumerable<PSResourceInfo> pkgsToInstall, // those found to be required to be installed (includes Dependency packages as well)
+            List<string> pckgNamesPassedInToInstall, // those requested by the user to be installed
             string repoName,
```
Comment (PaulHigin) [inline src/code/InstallHelper.cs]: Also, I notice that 'repoName' argument is not used.  Please remove.

**Review by PaulHigin — APPROVED** [review]
Also, please remove the unused 'repoName' parameter in the `private List<string> InstallPackage` method.

File: src/code/InstallHelper.cs (hunk 4 above)
Comment (PaulHigin) [reply src/code/InstallHelper.cs]: @anamnavi

Comment (anamnavi) [reply src/code/InstallHelper.cs]: I removed this

**Pull request merged**